From 72b390b0ebe9bc598b5ad24d6ed3f746171d66d9 Mon Sep 17 00:00:00 2001
From: Eric Sandeen <sandeen@redhat.com>
Date: Mon, 4 Aug 2014 19:03:53 -0400
Subject: [fs] btrfs: replace simple_strtoull() with kstrtoull()

Message-id: <1407179109-13714-347-git-send-email-sandeen@redhat.com>
Patchwork-id: 87242
O-Subject: [RHEL7.1 PATCH 346/422] btrfs: replace simple_strtoull() with kstrtoull()
Bugzilla: 1071179
RH-Acked-by: Zach Brown <zab@redhat.com>

From: ZhangZhen <zhenzhang.zhang@huawei.com>

commit 58dfae6365666e26cb2d172dc437773120fc4e1a
Author: ZhangZhen <zhenzhang.zhang@huawei.com>
Date:   Tue May 13 16:36:08 2014 +0800

    btrfs: replace simple_strtoull() with kstrtoull()

    use the newer and more pleasant kstrtoull() to replace simple_strtoull(),
    because simple_strtoull() is marked for obsoletion.

    Signed-off-by: Zhang Zhen <zhenzhang.zhang@huawei.com>
    Signed-off-by: Chris Mason <clm@fb.com>
Signed-off-by: Eric Sandeen <sandeen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/btrfs/ioctl.c b/fs/btrfs/ioctl.c
index a21a4ac..fba7a00 100644
--- a/fs/btrfs/ioctl.c
+++ b/fs/btrfs/ioctl.c
@@ -1503,11 +1503,12 @@ static noinline int btrfs_ioctl_resize(struct file *file,
  sizestr = vol_args->name;
  devstr = strchr(sizestr, ':');
  if (devstr) {
-  char *end;
   sizestr = devstr + 1;
   *devstr = '\0';
   devstr = vol_args->name;
-  devid = simple_strtoull(devstr, &end, 10);
+  ret = kstrtoull(devstr, 10, &devid);
+  if (ret)
+   goto out_free;
   if (!devid) {
    ret = -EINVAL;
    goto out_free;
-- 
1.7.1